bundle/regal/ast/keywords.rego

```diff
@@ -0,0 +1,132 @@
+package regal.ast
+
+import rego.v1
+
+keywords[row] contains keyword if {
+	some rule in input.rules
+
+	rule_text := base64.decode(rule.location.text)
+
+	some idx, line in split(rule_text, "\n")
+
+	col := indexof(line, " if ")
+	col > 0
+
+	row := rule.location.row + idx
+
+	not row in _comment_row_index
+
+	keyword := {
+		"name": "if",
+		"location": {
+			"row": rule.location.row + idx,
+			"col": col + 2,
+		},
+	}
+}
+
+keywords[pkg.location.row] contains keyword if {
+	pkg := input["package"]
+
+	keyword := {
+		"name": "package",
+		"location": {
+			"row": pkg.location.row,
+			"col": pkg.location.col,
+		},
+	}
+}
+
+keywords[imp.location.row] contains keyword if {
+	some imp in input.imports
+
+	keyword := {
+		"name": "import",
+		"location": {
+			"row": imp.location.row,
+			"col": imp.location.col,
+		},
+	}
+}
+
+keywords[loc.row] contains keyword if {
+	some rule in input.rules
+
+	loc := rule.head.location
+
+	col := indexof(base64.decode(loc.text), " contains ")
+
+	col > 0
+
+	keyword := {
+		"name": "contains",
+		"location": {
+			"row": loc.row,
+			"col": col + 2,
+		},
+	}
+}
+
+keywords[value.row] contains keyword if {
+	some rule in input.rules
+	some expr in rule.body
+
+	walk(expr.terms, [path, value])
```

Comment:
Probably good to add an `ast.exprs` similar to `ast.vars` at this point... as a few rules do this kind of walk now, and more likely to be added. It'd be nice to be able to reuse the result of that

Reply:
Hmm, I think I've managed to realise this in d8bd38bde5515c3088ce35a2be23d67c496e8969, but let me know if that's not what you had in mind. I had some trickiness with the rule indexing but it all works now. We just need to be careful to use the same meaning for rule index everywhere and it works!